### ISSUE TITLE
Enumerable#each_with_index is broken again

### ISSUE THREAD

**Issue by enebo** [jruby/jruby#4724]
```text
jruby -e '%i(foo bar baz).each_with_index {|v| puts v.inspect }
```

Prints out [:foo, 0], ... instead of :foo.

We broke this previous to 9.1.10.0 and then regressed again in .11/.12

Moved this over from #4610 so we can keep historical accuracy.

**Comment by enebo** [comment]
Fixed in commit d597e1a29571e648dbee5aadaec35b4a58c363f4 on master and cherry picked back to jruby-9.1.13.0 branch.  Not sure what happened here.  I am guessing I only reverted on 9.1.10.0 branch?  Will now see if a spec was only committed to that branch :|

**Comment by enebo** [comment]
Added regression 04dd4a78a0 spec.
